ibm/resource_ibm_cis_healthcheck.go

```diff
@@ -2,331 +2,364 @@ package ibm
-		if description, ok := d.GetOk("description"); ok {
-			monitorUpdate.Description = description.(string)
+		if description, ok := d.GetOk(cisGLBHealthCheckDesc); ok {
+			opt.SetDescription(description.(string))
 		}
-		if method, ok := d.GetOk("method"); ok {
-			monitorUpdate.Method = method.(string)
+		if method, ok := d.GetOk(cisGLBHealthCheckMethod); ok {
+			opt.SetMethod(method.(string))
 		}
-		if timeout, ok := d.GetOk("timeout"); ok {
-			monitorUpdate.Timeout = timeout.(int)
+		if timeout, ok := d.GetOk(cisGLBHealthCheckTimeout); ok {
+			opt.SetTimeout(int64(timeout.(int)))
 		}
-		if retries, ok := d.GetOk("retries"); ok {
-			monitorUpdate.Retries = retries.(int)
+		if retries, ok := d.GetOk(cisGLBHealthCheckRetries); ok {
+			opt.SetRetries(int64(retries.(int)))
 		}
-		if interval, ok := d.GetOk("interval"); ok {
-			monitorUpdate.Interval = interval.(int)
+		if interval, ok := d.GetOk(cisGLBHealthCheckInterval); ok {
+			opt.SetInterval(int64(interval.(int)))
 		}
-		if follow_redirects, ok := d.GetOk("follow_redirects"); ok {
-			monitorUpdate.FollowRedirects = follow_redirects.(bool)
+		if followRedirects, ok := d.GetOk(cisGLBHealthCheckFollowRedirects); ok {
+			opt.SetFollowRedirects(followRedirects.(bool))
 		}
-		if allow_insecure, ok := d.GetOk("allow_insecure"); ok {
-			monitorUpdate.AllowInsecure = allow_insecure.(bool)
+		if allowInsecure, ok := d.GetOk(cisGLBHealthCheckAllowInsecure); ok {
+			opt.SetAllowInsecure(allowInsecure.(bool))
 		}
-		if port, ok := d.GetOk("port"); ok {
-			monitorUpdate.Port = port.(int)
+		if port, ok := d.GetOk(cisGLBHealthCheckPort); ok {
+			opt.SetPort(int64(port.(int)))
 		}
-		_, err = cisClient.Monitors().UpdateMonitor(cisId, monitorId, monitorUpdate)
+		result, resp, err := sess.EditLoadBalancerMonitor(opt)
 		if err != nil {
-			log.Printf("[WARN] Error getting zone during MonitorUpdate %v \n", err)
+			log.Printf("Error updating global load balancer health check detail: %s", resp)
 			return err
 		}
+		log.Printf("Monitor update succesful : %s", *result.Result.ID)
 	}
-	return resourceCIShealthCheckRead(d, meta)
+
+	return resourceCISHealthCheckRead(d, meta)
 }
 
-func resourceCIShealthCheckDelete(d *schema.ResourceData, meta interface{}) error {
-	cisClient, err := meta.(ClientSession).CisAPI()
+func resourceCISHealthCheckDelete(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).CisGLBHealthCheckClientSession()
 	if err != nil {
 		return err
 	}
-	monitorId, cisId, err := convertTftoCisTwoVar(d.Id())
+
+	monitorID, crn, err := convertTftoCisTwoVar(d.Id())
 	if err != nil {
 		return err
 	}
-	var monitor *v1.Monitor
-	emptyMonitor := new(v1.Monitor)
-	monitor, err = cisClient.Monitors().GetMonitor(cisId, monitorId)
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewDeleteLoadBalancerMonitorOptions(monitorID)
+
+	result, resp, err := sess.DeleteLoadBalancerMonitor(opt)
 	if err != nil {
-		if checkCisMonitorDeleted(d, meta, err, monitor) {
```

Comment:
check for minitor delete and response not nil
Refer existing resource [here](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/resource_ibm_api_gateway_endpoint.go#L384)

Reply:
checkCisMonitorDeleted() is a dead code since I have added Exists() method